### ISSUE TITLE
Graph methods should give meaningful errors when vertices out of range

### ISSUE THREAD

**Issue by pmacg** [pmacg/py-sgtl#2]
Currently, if you ask for the volume of a set of vertices, but the set includes indices which are greater than the number of vertices in the graph, the code throws an `IndexError` from somewhere inside the graph code.

This is not totally unreasonable, but it would help debugging this kind of error if it gave a clear error message. Something like: "Vertex set includes indices larger than number of vertices".

**Comment by yonMaor** [comment]
Hey, can you eleborate a little further on this issue? I'd like to help with it :)

**Comment by pmacg** [comment]
Hi, that's great to hear! This library is very new and so there are many ways it could be improved with usability enhancements.

The minimal working example to demonstrate this issue is the following:

```
>>> import sgtl.graph
>>> graph = sgtl.graph.cycle_graph(5)
>>> vertices = [0, 1, 2, 3, 4, 5, 6]
>>> graph.volume(vertex_set)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/peter/.local/lib/python3.8/site-packages/sgtl/graph.py", line 101, in volume
    return sum([self.degrees[v] for v in vertex_set])
  File "/home/peter/.local/lib/python3.8/site-packages/sgtl/graph.py", line 101, in <listcomp>
    return sum([self.degrees[v] for v in vertex_set])
IndexError: list index out of range
```

This error happens because the constructed graph has `5` vertices (`[0, 1, 2, 3, 4]`) but the variable `vertices` contains indices which are outside of this range.

In my opinion, the error message above could be more helpful to the user. In particular, the error is raised at some internal point in the library code - the use should not need to understand the internal library code in order to debug their problem.

It would be useful to be told exactly that the set passed to `graph.volume()` has indices which do not correspond to vertices in the graph.

This probably affects more than just the `volume` method - it would be useful to have a look at the other methods on the `Graph` object as well.

Please let me know if I can help any further.

**Comment by yonMaor** [comment]
Just making sure that I'm understanding:
in your example you called:
`graph.volume(vertex_set)`

I'm assuming you meant:
`graph.volume(vertices)`

**Comment by pmacg** [comment]
Yes, I did!

**Comment by yonMaor** [comment]
I think I solve the issue, just a quick question: do you want the new message to be printed as an error, or as a normal print statement?

**Comment by pmacg** [comment]
Fixed by PR #12 .
